components/BottomNavigation/examples/MDCBottomNavigationBarController.m

```diff
@@ -0,0 +1,265 @@
+  }
+}
+
+/**
+ * Deselects the currently set item.  Sets the selectedIndex to NSNotFound, the naviagation bar's
+ * selected item to nil, and the selectedViewController to nil.
+ */
+- (void)deselectCurrentItem {
+  _selectedIndex = NSNotFound;
+  _navigationBar.selectedItem = nil;
+
+  // Force removal of the currently selected viewcontroller if there is one.
+  self.selectedViewController = nil;
+}
+
+/**
+ * Sets the selected view controller to the corresponding index and updates the navigation bar's
+ * selected item.
+ */
+- (void)updateViewsForSelectedIndex:(NSUInteger)index {
+  // Update the selected view controller
+  UIViewController *selectedViewController = [_viewControllers objectAtIndex:index];
+  self.selectedViewController = selectedViewController;
+
+  // Update the navigation bar's selected item.
+  _navigationBar.selectedItem = selectedViewController.tabBarItem;
+  [self setNeedsStatusBarAppearanceUpdate];
+}
+
+/**
+ * Hooks up the constraints for the subviews of this controller.  Namely the content view and the
+ * navigation bar.
+ */
+- (void)loadConstraints {
+  _content.translatesAutoresizingMaskIntoConstraints = NO;
+  _navigationBar.translatesAutoresizingMaskIntoConstraints = NO;
+
+  // Navigation Bar Constraints
+  [self.view.leftAnchor constraintEqualToAnchor:_navigationBar.leftAnchor].active = YES;
+  [self.view.rightAnchor constraintEqualToAnchor:_navigationBar.rightAnchor].active = YES;
+
+  _navigationBarHeightConstraint =
+      [_navigationBar.heightAnchor constraintEqualToConstant:[self calculateNavigationBarHeight]];
+  _navigationBarHeightConstraint.active = YES;
+
+  [_navigationBar.topAnchor constraintEqualToAnchor:_content.bottomAnchor].active = YES;
+  [_navigationBar.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor].active = YES;
+
+  [_navigationBar setContentHuggingPriority:UILayoutPriorityDefaultHigh
+                                    forAxis:UILayoutConstraintAxisHorizontal];
+
+  // Content View Constraints
+  [self.view.leftAnchor constraintEqualToAnchor:_content.leftAnchor].active = YES;
+  [self.view.rightAnchor constraintEqualToAnchor:_content.rightAnchor].active = YES;
+
+  [self.view.topAnchor constraintEqualToAnchor:_content.topAnchor].active = YES;
+}
+
+/**
+ * Pins the given view to the edges of the content view.
+ */
+- (void)addConstraintsForContentView:(UIView *)view {
+  view.translatesAutoresizingMaskIntoConstraints = NO;
+  [view.leadingAnchor constraintEqualToAnchor:_content.leadingAnchor].active = YES;
+  [view.trailingAnchor constraintEqualToAnchor:_content.trailingAnchor].active = YES;
+  [view.topAnchor constraintEqualToAnchor:_content.topAnchor].active = YES;
+  [view.bottomAnchor constraintEqualToAnchor:_content.bottomAnchor].active = YES;
+}
+
+/** Returns the desired height of the navigation bar. **/
+- (CGFloat)calculateNavigationBarHeight {
+  return [_navigationBar sizeThatFits:self.view.bounds.size].height;
```

Comment:
Please calculate the `CGSize` with `CGRectGetHeight` and `CGRectGetWidth` instead of accessing the bounds directly.  It's possible for the bounds to be non-standardized with a negative size.

Reply:
Done.